www/js/view_single.js

```diff
@@ -1,45 +1,56 @@
-$(document).ready( function() {
-    $(document).on('change', '.btn-file :file', function() {
-        var input = $(this),
-        label = input.val().replace(/\\/g, '/').replace(/.*\//, '');
-        input.trigger('fileselect', [label]);
-        });
 
-        $('.btn-file :file').on('fileselect', function(event, label) {
-            
-            var input = $(this).parents('.input-group').find(':text'),
-                log = label;
-            
-            if( input.length ) {
-                input.val(log);
-            } else {
-                if( log ) alert(log);
-            }
-        
-    });
-    function readURL(input) {
-        if (input.files && input.files[0]) {
-            var reader = new FileReader();
-            
-            reader.onload = function (e) {
-                $('#img-upload').attr('src', e.target.result);
-            }
-            
-            reader.readAsDataURL(input.files[0]);
-        }
-    }
 
-    $("#imgInp").change(function(){
-        readURL(this);
-    });   
+$(document).ready( function() {
 
     lightbox.option({
       'resizeDuration': 200,
       'wrapAround': true
-    })
+    });
     
 });
 
+Dropzone.autoDiscover = false;
+
+var myDropzone = new Dropzone("#form-upload", {
+    paramName: "file", // The name that will be used to transfer the file
+    maxFilesize: 2, // MB
+    uploadMultiple : true, 
+    parallelUploads: 0,
+    autoProcessQueue: false,
+    acceptedFiles: 'image/*'
+});
+
+myDropzone.on("addedfile", function(file) { 
+    $.ajax({
+        url: '/generate_upload_url',
```

Comment:
I guess this is the only thing I don't understand, why is this code really similar to the one in `startUpload()`?
Is this one used in a code path that I'm not seeing?

Reply:
Dropzone is doing parallel uploads no matter what I tell it, for some reason. So in order to cover for every single upload having a fresh url (since it's the requirement for the blobstore) I generate the url before and after processing.

Reply:
Just wondering, which is before and which is after?

Reply:
on."addedfile" is triggered right after the file was successfully sent to the back. 

My processUploads function iterates through the Queue generating a url before the file is even processed, so this one comes first.